packages/rn-tester/js/examples/TextInput/TextInputSharedExamples.js

```diff
@@ -122,18 +120,40 @@ class RewriteExampleInvalidCharacters extends React.Component<
 > {
   constructor(props) {
     super(props);
-    this.state = {text: ''};
+    this.state = {text: '', selection: {start: -1, end: -1}};
   }
+
+  onSelectionChangeHandler = ({nativeEvent: {selection}}) => {
+    const {start, end} = selection;
+    const {text} = this.state;
+    const maxEnd = text.length;
+    const validEnd = end > maxEnd ? maxEnd : end;
+    const validStart = start > maxEnd ? maxEnd : start;
+    const newSelection = {start: validStart, end: validEnd};
+    this.setState({selection: newSelection});
+  };
+
   render() {
+    const {
+      text,
+      selection: {start, end},
+    } = this.state;
     return (
       <View style={styles.rewriteContainer}>
         <TextInput
           testID="rewrite_no_sp_input"
           autoCorrect={false}
-          multiline={false}
+          multiline={true}
           onChangeText={text => {
-            this.setState({text: text.replace(/\s/g, '')});
+            const newText = text.replace(/\s/g, '');
```

Comment:
as this component keeps controller version of the text in JavaScript, the text value `an example` would differ in javascript after the above substitution and become `anexample`, the onSelectionOnChangeHandler will control the cursor position and avoid setting it to a position higher then the new word length.

This behaviour is different between iOS and Android and causes some issues. I will try to remove the `onSelectionOnChangeHandler` that controls the cursor position.